internal/commands/inspect_builder.go

```diff
@@ -266,35 +272,152 @@ func runImagesOutput(runImage string, mirrors []string, cfg config.Config) (stri
 	return strings.TrimSuffix(buf.String(), "\n"), nil
 }
 
-func detectionOrderOutput(order dist.Order) (string, error) {
+type stackEntry struct {
+	LayerInfo dist.BuildpackRef
+	Depth     int
+	Last      bool
+}
+
+func reverseStack(stack []stackEntry) []stackEntry {
+	left := 0
+	right := len(stack) - 1
+	for left < right {
+		stack[left], stack[right] = stack[right], stack[left]
+		left++
+		right--
+	}
+	return stack
+}
+
+func detectionOrderOutput(order dist.Order, layers dist.BuildpackLayers, maxDepth int) (string, error) {
 	buf := strings.Builder{}
-	for i, group := range order {
-		buf.WriteString(fmt.Sprintf("  Group #%d:\n", i+1))
-
-		tabWriter := new(tabwriter.Writer).Init(&buf, 0, 0, 4, ' ', 0)
-		for _, bp := range group.Group {
-			var optional string
-			if bp.Optional {
-				optional = "(optional)"
-			}
+	tabWriter := new(tabwriter.Writer).Init(&buf, 0, 0, 4, ' ', 0)
```

Comment:
I understand this was the same as what we had before, but any chance we can have a note explaining these magic numbers?

Reply:
Feel like pulling these out into 'consts' in this file would work? They're used in a couple other places